src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java

```diff
@@ -94,39 +95,73 @@ public void showError(String msg) {
         Notification.show(msg);
     }
 
-    public void showSaveNotification(String msg) {
+    /**
+     * shows a message in a notification.
+     *
+     * @param msg
+     */
+    public void showNotification(String msg) {
         Notification.show(msg);
     }
 
+    /**
+     * Enables/Disables the new product button.
+     *
+     * @param enabled
+     */
     public void setNewProductEnabled(boolean enabled) {
         newProduct.setEnabled(enabled);
     }
 
+    /**
+     * Clears the fields in the product form.
+     */
     public void clearSelection() {
         grid.getSelectionModel().deselectAll();
     }
 
+    /**
+     * Selects a row
+     * 
+     * @param row
+     */
     public void selectRow(Product row) {
         grid.getSelectionModel().select(row);
     }
 
-    public Product getSelectedRow() {
-        return grid.getSelectedRow();
-    }
-
+    /**
+     * Updates a product
+     * 
+     * @param product
+     */
     public void updateProduct(Product product) {
         dataProvider.save(product);
     }
 
+    /**
+     * Removes a product
```

Comment:
I still feel this should explain "removed where". Like "removes the Product from 
(commented in #42) Same goes for update, edit, select.